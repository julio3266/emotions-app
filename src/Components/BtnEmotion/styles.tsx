import styled from 'styled-components/native';

export const Button = styled.TouchableOpacity`
  background-color: ${({theme}) => theme.colors.darkTransparency};
  justify-content: center;
  align-items: center;
  width: 80px;
  height: 80px;
  border-radius: 50px;
  margin-left: 2%;
  margin-top: 2%;
`;

export const Title = styled.Text`
  color: ${({theme}) => theme.colors.white};
  font-family: ${({theme}) => theme.fonts.robotoRegular};
  font-size: 15px;
  font-weight: 500;
  text-align: center;
`;

export const Emoji = styled.Text`
  font-family: ${({theme}) => theme.fonts.robotoRegular};
  text-align: center;
`;
