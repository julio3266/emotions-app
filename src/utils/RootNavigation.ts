import {createRef} from 'react';

export const navigationRef: any = createRef();

export const navigate = (name: string, params?: any) => {
  navigationRef.current?.navigate(name, params);
};

export const goBack = () => {
  navigationRef.current?.goBack();
};

export const params = () => {
  navigationRef.current?.params;
};
