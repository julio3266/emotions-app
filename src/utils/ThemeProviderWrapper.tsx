import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import Theme from '../theme/styles';

interface IProps {
  children: any;
  padding?: boolean;
}

export const ThemeProviderWrapper: React.FC<IProps> = ({children}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};
