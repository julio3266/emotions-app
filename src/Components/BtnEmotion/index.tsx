import React from 'react';
import {assignTestId} from '@utils/qualityAssurance';
import * as Styled from './styles';

export interface IBtnEmotionProps {
  testID: string;
  title?: string;
  emoji?: any;
  handlePress?: () => void;
}

export const BtnEmotion: React.FC<IBtnEmotionProps> = ({
  testID,
  title,
  emoji,
  handlePress,
}) => {
  return (
    <Styled.Button
      onPress={handlePress}
      {...assignTestId('TouchableOpacity', testID)}>
      <Styled.Title {...assignTestId('View', `${testID}_title`)}>
        {title}
      </Styled.Title>
      <Styled.Emoji {...assignTestId('View', `${testID}_emoji`)}>
        {emoji}
      </Styled.Emoji>
    </Styled.Button>
  );
};
