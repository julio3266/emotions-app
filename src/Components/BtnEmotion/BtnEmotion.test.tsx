import React from 'react';
import {render} from '@testing-library/react-native';
import {mockTestID} from '../../utils/qualityAssurance';
import {ThemeProviderWrapper as wrapper} from '../../utils/ThemeProviderWrapper';
import {BtnEmotion, IBtnEmotionProps} from '@Components/BtnEmotion';

describe('Components/BtnEmotion', () => {
  let props: IBtnEmotionProps = {
    testID: 'BtnEmotion',
    emoji: '#x1F603',
    handlePress: jest.fn(),
    title: 'Happy',
  };
  const {findByTestId, toJSON} = render(<BtnEmotion {...props} />, {
    wrapper,
  });

  const sut_container = mockTestID('TouchableOpacity', props.testID!).testID;
  const sut_title = mockTestID('Text', `${props.testID!}_title`).testID;
  const sut_emoji = mockTestID('Text', `${props.testID!}_emoji`).testID;
  it(`should render #${sut_container}`, () => {
    const sut = findByTestId(sut_container);
    expect(sut).toBeTruthy();
  });

  it(`should render #${sut_title}`, () => {
    const sut = findByTestId(sut_title);
    expect(sut).toBeTruthy();
  });

  it(`should render #${sut_emoji}`, () => {
    const sut = findByTestId(sut_emoji);
    expect(sut).toBeTruthy();
  });

  it(`should render a snapshot`, () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
