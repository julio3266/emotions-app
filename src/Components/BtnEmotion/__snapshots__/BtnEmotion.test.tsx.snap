// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/BtnEmotion should render a snapshot 1`] = `null`;
